src/silicone/utils.py

```diff
@@ -181,12 +181,21 @@ def _remove_t0_from_wide_db(times_needed, _db):
             _db.loc[model, scenario, time] = _db.loc[model, scenario, time] - offset
 
 
-def _make_interpolator(variable_follower, variable_leader, wide_db, time_col):
+def _make_interpolator(
+    variable_follower, variable_leader, wide_db, time_col, interpkind="linear"
+):
     """
     Constructs a linear interpolator for variable_follower as a function of
     (one) variable_leader for each timestep in the data.
     """
     derived_relationships = {}
+    # We want a generic function to return constants where required
+    def f_factory(ys):
```

Comment:
Move to a private function? I can't see why this function is nested but maybe I'm missing something